candle-transformers/src/models/stable_diffusion/uni_pc.rs

```diff
@@ -0,0 +1,1048 @@
+//! # UniPC Scheduler
+//!
+//! UniPCMultistepScheduler is a training-free framework designed for
+//! fast sampling of diffusion models.
+//!
+//! It consists of a corrector (UniC) and a predictor (UniP) that share a unified analytical form and
+//! support arbitrary orders. UniPC is by design model-agnostic, supporting pixel-space/latent-space
+//! DPMs on unconditional/conditional sampling. It can also be applied to both noise prediction and
+//! data prediction models. The corrector UniC can be also applied after any off-the-shelf solvers
+//! to increase the order of accuracy.
+//!
+//! UniPC: A Unified Predictor-Corrector Framework for Fast Sampling of Diffusion Models, W. Zhao et al, 2023.
+//! https://arxiv.org/abs/2302.04867
+//!
+//! This work is based largely on the diffusers implementation: https://github.com/huggingface/diffusers
```

Comment:
If there is an equivalent file in `diffusers` could we have a direct permalink to it so that it's easier to compare with the version that was ported?

Reply:
Sure, updated in 2204ef8d1ddcbb588d5c764a94d91da4883e47d4

This deviates from that implementation a bit in terms of how the schedule is specified since I was mainly interested in the Karras schedule. Extending it to support all the options available there shouldn't be too difficult though if that's preferred.